File: typescript/Behavioral/13-ChainOfResponsibility/Handler/HandymanApprover.ts
import { BaseApprovalHandler } from "./BaseApprovalHandler";
import { ApprovalRequest } from "../Model/ApprovalRequest";

export class HandymanApprover extends BaseApprovalHandler {
  private static readonly LIMIT = 1_000;

  handle(request: ApprovalRequest): void {
    if (request.cost <= HandymanApprover.LIMIT)
      console.log(`  ✅ Handyman approved: ${request}`);
    else {
      console.log(
        `  ↗️  Handyman escalating (N$${request.cost.toFixed(2)} > N$${HandymanApprover.LIMIT.toFixed(2)})...`,
      );
      super.handle(request);
    }
  }
}

File: csharp/Behavioral/13-ChainOfResponsibility/Handler/HandymanApprover.cs
public class HandymanApprover : BaseApprovalHandler
{
    private const decimal Limit = 1_000m;

    public override void Handle(ApprovalRequest request)
    {
        if (request.Cost <= Limit)
            Console.WriteLine($"  ✅ Handyman approved: {request}");
        else
        {
            Console.WriteLine($"  ↗️  Handyman escalating (N${request.Cost:N2} > N${Limit:N2})...");
            base.Handle(request);
        }
    }
}
